src/likelihoods/studentt.jl

```diff
@@ -0,0 +1,29 @@
+
+using SpecialFunctions :: logbeta
+using IrrationalConstants :: logπ, 
+
+
+"""
+    StudentTLikelihood(σ²,ν)
+
+Student's T likelihood with `σ²` scale and ν degrees of freedom . This is to be used if we assume that the
+uncertainity associated with the data follows a Student's T distribution.
+
+```math
+    p(y|f) = \\operatorname{Student}(y | f, σ², ν)
+```
+"""
+
+struct StudentTLikelihood{T<:Real, Tn :: Real} <: AbstractLikelihood
+    σ²::Vector{T}
+    ν::Vector{Tn}
```

Comment:
We tend to change the implementation to avoid storing everything in Vector (I know it's not true for `GaussianLikelihood` but it should change).
```suggestion
    σ²::T
    ν::Tn
```